apluslms_roman/cli.py

```diff
@@ -365,20 +416,136 @@ def build_action(context):
+
+    for val in context.args.values:
+        try:
+            key, val = val.split('=', 1)
+            document.mlset_cast(key, val)
+        except ValueError as err:
+            if hasattr(err, 'value_type'):
+                raise ValueError(_("{} should be of type '{}', but was '{}'.").format(
+                    key, err.value_type, type(val).__name__))
+            else:
+                raise ValueError(_("Give values in format 'key=val'."))
+
+    document.validate()
+    document.save()
+
+
+def config_rm_action(context):
+    if context.args.global_ == context.args.project:
+        raise Exception("Choose either global settings or project settings")
+    document = context.settings if context.args.global_ else get_config(context)
+
+    for val in context.args.keys:
+        document.mldel(val)
+
+    document.validate()
+    document.save()
+
+
+def step_list_action(context):
+    steps = get_config(context).steps
+    num_len = max(2, len(str(len(steps)-1)))
+    name_len = max(4, max(len(s.get('name', '')) for s in steps))
+    header_fmt = "{:>%ds}  {:%ds} {}" % (num_len, name_len)
+    step_fmt = "{:%dd}. {:%ds} {}" % (num_len, name_len)
+    print(header_fmt.format('ID', 'NAME', 'IMAGE'))
+    for i, step in enumerate(steps):
+        print(step_fmt.format(i, step.get('name', ''), step['img']))
+    return
+
+
+def step_add_action(context):
+    args = context.args
+    env = args.env
+    if env:
+        try:
+            env = {a: b for a, b in [s.split('=', 1) for s in env]}
+        except ValueError:
+            raise ValueError("env is a dict, so values need to be in key=val format, e.g. a=1 b=2")
+    step = {
+        'img': args.img,
+        'cmd': args.cmd,
+        'mnt': args.mnt,
+        'env': env,
+        'name': args.name
+    }
+    step = {k: v for k, v in step.items() if v}
+    config = get_config(context)
+    config.add_step(step)
+    try:
+        config.validate()
+    except ValidationError as err:
+        exit(1, '\n'.join(render_error(err)))
+    config.save()
+    print("Step successfully added to config.")
+
+
+def step_rm_action(context):
+    def confirm_del(step):
+        print('step:')
+        print('  {}'.format(yaml_dump(step.get_data()).replace('\n', '\n  ')))
+        while True:
+            i = input("Delete this step? (y/n) ").lower()
+            if i == 'y':
+                return True
+            if i == 'n':
+                return False
+
+    ref = context.args.ref.lower()
+    config = get_config(context)
+    try:
+        step = config.get_step(ref)
+    except IndexError:
+        raise IndexError(_(
+            "Index is out of range. Remember, indexing start from 0. " +
+            "Use 'roman --list-steps' to see step indexes."))
+    except KeyError:
+        raise KeyError(_("There is no step called '{}'").format(ref))
+    if not context.args.force and not confirm_del(step):
+        return
+    config.del_step(ref)
```

Comment:
It would be nice if the step dict could be passed to `del_step` too (string/int ref is also important interface). Current solution is fine for now. Just thinking aloud,

Reply:
well I changed `del_step` so it allows dicts too, didn't really require much. (oops forgot to send this comment earlier)